src/librustc_parse/parser/mod.rs

```diff
@@ -897,12 +891,20 @@ impl<'a> Parser<'a> {
 
     /// Parses unsafety: `unsafe` or nothing.
     fn parse_unsafety(&mut self) -> Unsafe {
-        if self.eat_keyword(kw::Unsafe) { Unsafe::Yes(self.prev_span) } else { Unsafe::No }
+        if self.eat_keyword(kw::Unsafe) {
+            Unsafe::Yes(self.normalized_prev_token.span)
+        } else {
+            Unsafe::No
+        }
     }
 
     /// Parses constness: `const` or nothing.
     fn parse_constness(&mut self) -> Const {
-        if self.eat_keyword(kw::Const) { Const::Yes(self.prev_span) } else { Const::No }
+        if self.eat_keyword(kw::Const) {
+            Const::Yes(self.normalized_prev_token.span)
```

Comment:
And this is also because of `eat_keyword` dealing with identifiers? Although `check_keyword` doesn't seem to be using `normalized_token`.

Reply:
Same answer as in https://github.com/rust-lang/rust/pull/69579#discussion_r386033468, `const` is an identifier.

`eat_keyword` works correctly because it uses `token.ident()` internally which uses normalization.